server/src/main/java/org/tctalent/server/service/db/CandidateExamService.java

```diff
@@ -46,4 +46,7 @@ CandidateExam createExam(
      * @return List of updated candidate exam records
      */
     CandidateExam updateCandidateExam(UpdateCandidateExamRequest request);
+
+    List<CandidateExam> list(long id);
```

Comment:
I meant something like this. It should also appear on the implementation.

```suggestion
    @NotNull
    List<CandidateExam> list(long id);
```

Also it would be good to have some JavaDoc for this method. The JavaDoc could let the user know what the id is (a candidate id I assume) and what happens when the id is not associated with a known candidate.

Reply:
The @NotNull also should appear on the implementation of the list method - unfortunately implementations of interfaces do not automatically copy annotations.